src/logcollector/read_audit.c

```diff
@@ -19,19 +19,19 @@ static void audit_send_msg(char **cache, int top, const char *file, int drop_it,
     int i;
     size_t n = 0;
     size_t z;
-    char message[OS_MAXSTR];
+    char message[OS_MAX_LOG_SIZE] = {0};
 
     for (i = 0; i < top; i++) {
         z = strlen(cache[i]);
 
-        if (n + z + 1 < OS_MAXSTR) {
+        if (n + z + 1 < sizeof(message)) {
             if (n > 0)
                 message[n++] = ' ';
 
-            strncpy(message + n, cache[i], z);
+            strncat(message, cache[i], OS_MAX_LOG_SIZE - 1 - n);
```

Comment:
Using `strncat` here will bump up the time complexity of this function to O(n^2), because every call to `strncat` will have to find the null terminated on `message`. 
I think it would be better to just keep using `strncpy` here like so:
```
strncpy(message + n, cache[i], sizeof message - 1 - n);
```
As an extra, we could either print a warning or crash in the "else" condition of this `if`. Because that would mean we ran out of space in `message`.
@Dwordcito @jnasselle: mentioning you both to make sure I didn't miss anything here.

Reply:
I see. That's a good point. Maybe we could keep using `strncat` but pass `message + n` as the target and the remaining space left on it.